rotkehlchen/chain/optimism/modules/llamazip/decoder.py

```diff
@@ -0,0 +1,27 @@
+from typing import TYPE_CHECKING, Final
+
+from rotkehlchen.chain.evm.decoding.llamazip.decoder import LlamazipCommonDecoder
+from rotkehlchen.chain.evm.types import string_to_evm_address
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.ethereum.node_inquirer import EthereumInquirer
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.user_messages import MessagesAggregator
+
+ROUTER_ADDRESS: Final = string_to_evm_address('0x6f9d14Cf4A06Dd9C70766Bd161cf8d4387683E1b')
+
+
+class LlamazipDecoder(LlamazipCommonDecoder):
+
+    def __init__(
+            self,
+            ethereum_inquirer: 'EthereumInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+    ) -> None:
+        super().__init__(
+            evm_inquirer=ethereum_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+            router_address=ROUTER_ADDRESS,
```

Comment:
check what left mentioned about different addresses on arbitrum. YOu can provide a list and then in the address to counterparties do something like

dict.fromkeys(self.addresses, (CPT_LLAMAZIP,))

I'm not sure if ^ that will be it but something similar